tests/controllers/qos_controller_test.cpp

```diff
@@ -37,8 +41,13 @@ TEST(SerenityControllerTest, NoQoSCorrections) {
   Try<Nothing> initialize = controller->initialize(
       lambda::bind(&MockSlaveUsage::usage, &usage));
 
+  process::Clock::pause();
```

Comment:
If you deleted the leaked controller, would that not terminate and wait for the process correctly?
Also check out the `EXPECT_CALL(<process>, <function>).WillOnce(Return(<expected-return-value>));` macros. Maybe this will help you develop further tests!